src/brush.rs

```diff
@@ -60,6 +76,29 @@ impl<'a> BrushRef<'a> {
             Self::Image(image) => Brush::Image((*image).clone()),
         }
     }
+
+    /// Returns the brush with the alpha component multiplied by the specified
+    /// factor.
+    #[must_use]
+    pub fn with_alpha_factor(&self, alpha: f32) -> Brush {
```

Comment:
If we make the change suggested above, this one might not be needed. I think it’s probably fine to call `to_owned().with_alpha_factor()` on a `BrushRef`.